### PR TITLE
PLT-3176 On receive of a comment that we don't have the thread for, acquire the thread

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
When another user commented on an unloaded thread, undefined behaviour occurred. This PR updates the client to acquire missing threads for new comments from the server to fix the issue.

Also fixed a `#` getting added to the URL from clicking "Load more messages"

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3176

### CODE REVIEW

File: webapp/actions/post_actions.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import PostStore from 'stores/post_store.jsx';
+import TeamStore from 'stores/team_store.jsx';
+
+import Constants from 'utils/constants.jsx';
+const ActionTypes = Constants.ActionTypes;
+
+import Client from 'utils/web_client.jsx';
+import * as AsyncClient from 'utils/async_client.jsx';
+
+export function switchFromLdapToEmail(email, password, ldapPassword, onSuccess, onError) {
```
Comment (coreyhulen) [inline webapp/actions/post_actions.jsx]: why is this in post_actions.jsx?

**Comment by coreyhulen** [comment]
minor comment otherwise +1

File: webapp/actions/post_actions.jsx (hunk 1 above)
Comment (jwilander) [inline webapp/actions/post_actions.jsx]: That's a mistake, not sure how I missed that. Will fix

**Comment by crspeller** [comment]
+1

**Pull request merged**